kernel-kmp-script

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+op=$(echo "$0" | sed -e "s/.*-//")
+
+while true ; do
+    case $1 in
+	--version)
+	    version="$2"
+	    shift 2
+	    ;;
+	--release)
+	    release="$2"
+	    shift 2
+	    ;;
+	--name)
+	    name="$2"
+	    shift 2
+	    ;;
+
+	--kernelrelease)
+	    kernelrelease="$2"
+	    shift 2
+	    ;;
+	--flavor)
+	    flavor="$2"
+	    shift 2
+	    ;;
+	--usrmerged)
+	    usrmerged="$2"
+	    shift 2
+	    ;;
+
+	*) break
+	    ;;
+    esac
+done
+
+wm2=/usr/lib/module-init-tools/weak-modules2
+nvr="$name"-"$version"-"$release"
+
+run_wm2() {
+    [ -z "$KERNEL_PACKAGE_SCRIPT_DEBUG" ] || echo wm2 "$@" > /dev/kmsg
+    /bin/bash -${-/e/} $wm2 "$@"
+}
+
+[ -z "$KERNEL_PACKAGE_SCRIPT_DEBUG" ] || \
+    echo KMP "$op" name: "$name" version: "$version" release: "$release" \
+    kernelrelease: "$kernelrelease" flavor: "$flavor" \ usrmerged: "$usrmerged" > /dev/kmsg
+
+script_rc=0
+
+case $op in
+    pre)
+	;;
+    preun)
+	rpm -ql "$nvr" | sed -n '/\.ko\(\.xz\|\.gz\|\.zst\)\?$/p' > "/var/run/rpm-$nvr-modules" || script_rc=$?
+	;;
+    post)
+	if [ -x $wm2 ]; then
+	    INITRD_IN_POSTTRANS=1 run_wm2 --add-kmp $nvr
+	fi
+	;;
+    postun)
+	modules=( $(cat /var/run/rpm-$nvr-modules) )
+	rm -f "/var/run/rpm-$nvr-modules"
+	if [ ${#modules[*]} = 0 ]; then
+	    echo "WARNING: $nvr does not contain any kernel modules" >&2
+	    exit 0
+	fi
+	if [ -x $wm2 ]; then
+	    printf '%s\n' "${modules[@]}" | run_wm2 --remove-kmp $nvr || script_rc=$?
+	fi
+	;;
+    posttrans)
+	eval "$(rpm -E "%{?regenerate_initrd_posttrans}")" || script_rc=$?
```

Comment:
Ugh. Please let's just copy the code from the macro here. Is there any reason not to do that?

Reply:
The macro can be defined differently in different codestreams?

It's just what the script does currently in the kernel translated to shell.

Reply:
> The macro can be defined differently in different codestreams?

Most certainly not. We move this functionality from rpm scriptlets to regular scripts now. Such scripts shouldn't need to run `rpm -E`. If we ever get a code stream that installs `regenerate_initrd_posttrans` elsewhere, we can change the path directly in the code (note that `r-i-p` is part of *s-m-t* itself).